dataline-commons/src/test/java/io/dataline/commons/logging/Log4j2ConfigTest.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Dataline
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.dataline.commons.logging;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import io.dataline.commons.io.IOs;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+public class Log4j2ConfigTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Log4j2ConfigTest.class);
+
+  private Path root;
+
+  @BeforeEach
+  void setUp() throws IOException {
+    root = Files.createTempDirectory("test");
+    MDC.clear();
+  }
+
+  @Test
+  void testWorkerDispatch() {
+    MDC.put("context", "worker");
+    MDC.put("job_root", root.toString());
+    MDC.put("job_id", "1");
+
+    LOGGER.error("random message");
+
+    assertTrue(IOs.readFile(root, "logs.log").contains("random message"));
```

Comment:
should this test that "worker" and "job-id" are in the output logs as well?

Reply:
not really necessary. what matter is that the message is in the scoped file